core/trino-main/src/main/java/io/trino/operator/function/TableFunctionOperator.java

```diff
@@ -285,21 +294,36 @@ public TableFunctionOperator(
         PagesIndex pagesIndex = pagesIndexFactory.newPagesIndex(sourceTypes, expectedPositions);
         HashStrategies hashStrategies = new HashStrategies(pagesIndex, partitionChannels, prePartitionedChannels, sortChannels, sortOrders, preSortedPrefix);
 
-        this.outputPages = pageBuffer.pages()
-                .transform(new PartitionAndSort(pagesIndex, hashStrategies, processEmptyInput))
-                .flatMap(groupPagesIndex -> pagesIndexToTableFunctionPartitions(
-                        groupPagesIndex,
-                        hashStrategies,
-                        tableFunctionProvider,
-                        session,
-                        functionHandle,
-                        properChannelsCount,
-                        passThroughSourcesCount,
-                        requiredChannels,
-                        markerChannels,
-                        passThroughSpecifications,
-                        processEmptyInput))
-                .flatMap(TableFunctionPartition::toOutputPages);
+        if (isRowSemantics && passThroughSpecifications.isEmpty()) {
+            StreamingWorkProcessor streamingWorkProcessor = new StreamingWorkProcessor(processEmptyInput, getOnlyElement(requiredChannels));
+            this.outputPages = pageBuffer.pages()
+                    .transform(streamingWorkProcessor.toTableFunctionInput())
+                    .transform(new TableFunctionWorkProcessor(tableFunctionProvider.getDataProcessor(session, functionHandle)))
+                    .map(page -> {
+                        Block[] blocks = new Block[properChannelsCount];
```

Comment:
nit: This lambda could be a method like `Page.getFirstNColumns`

Reply:
I don't think we need them today - as we don't support passthrough yet